x-pack/filebeat/input/awss3/input.go

```diff
@@ -115,9 +115,12 @@ func (in *s3Input) Run(inputContext v2.Context, pipeline beat.Pipeline) error {
 	defer cancelInputCtx()
 
 	if in.config.QueueURL != "" {
-		regionName, err := getRegionFromQueueURL(in.config.QueueURL, in.config.AWSConfig.Endpoint)
-		if err != nil {
-			return fmt.Errorf("failed to get AWS region from queue_url: %w", err)
+		regionName := in.config.RegionName
+		if regionName == "" {
+			regionName, err = getRegionFromQueueURL(in.config.QueueURL, in.config.AWSConfig.Endpoint)
```

Comment:
How about some unit tests around this?

Reply:
The existing code here is complected, so making a unit test here is not feasible without a refactor.

Reply:
Ok. We can consider as part of improvements.